src/NexusMods.Collections/InstallCollectionJob.cs

```diff
@@ -0,0 +1,218 @@
+using System.Collections.Concurrent;
+using System.Text.Json;
+using DynamicData.Kernel;
+using Microsoft.Extensions.DependencyInjection;
+using NexusMods.Abstractions.Collections;
+using NexusMods.Abstractions.Collections.Json;
+using NexusMods.Abstractions.Collections.Types;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.IO;
+using NexusMods.Abstractions.Jobs;
+using NexusMods.Abstractions.Library;
+using NexusMods.Abstractions.Library.Models;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.NexusModsLibrary;
+using NexusMods.Abstractions.NexusWebApi.Types;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Networking.NexusWebApi;
+using NexusMods.Paths;
+namespace NexusMods.Collections;
+
+using ModInstructions = (Mod Mod, LibraryFile.ReadOnly LibraryFile);
+
+
+public class InstallCollectionJob : IJobDefinitionWithStart<InstallCollectionJob, NexusCollectionLoadoutGroup.ReadOnly>
+{
+    private IJobContext<InstallCollectionJob> _context = null!;
+    public required NexusModsCollectionLibraryFile.ReadOnly SourceCollection { get; init; }
+    
+    public required IFileStore FileStore { get; init; }
+    
+    public required JsonSerializerOptions JsonSerializerOptions { get; init; }
+    public required ILibraryService LibraryService { get; set; }
+    public required IConnection Connection { get; set; }
+    public required NexusModsLibrary NexusModsLibrary { get; set; }
+    
+    public required TemporaryFileManager TemporaryFileManager { get; set; }
+    
+    public required LoadoutId TargetLoadout { get; set; }
+
+
+
+
+    public static IJobTask<InstallCollectionJob, NexusCollectionLoadoutGroup.ReadOnly> Create(IServiceProvider provider, LoadoutId target, NexusModsCollectionLibraryFile.ReadOnly source)
+    {
+        var monitor = provider.GetRequiredService<IJobMonitor>();
+        var job = new InstallCollectionJob
+        {
+            TargetLoadout = target,
+            SourceCollection = source,
+            FileStore = provider.GetRequiredService<IFileStore>(),
+            JsonSerializerOptions = provider.GetRequiredService<JsonSerializerOptions>(),
+            LibraryService = provider.GetRequiredService<ILibraryService>(),
+            NexusModsLibrary = provider.GetRequiredService<NexusModsLibrary>(),
+            Connection = provider.GetRequiredService<IConnection>(),
+            TemporaryFileManager = provider.GetRequiredService<TemporaryFileManager>(),
+        };
+        return monitor.Begin<InstallCollectionJob, NexusCollectionLoadoutGroup.ReadOnly>(job);
+    }
+
+
+
+    public async ValueTask<NexusCollectionLoadoutGroup.ReadOnly> StartAsync(IJobContext<InstallCollectionJob> context)
+    {
+        _context = context;
```

Comment:
I don't think we should be saving the context like this.

Reply:
Does the context even need to be saved? It seems only the cancellation token is used out of it.
And you can pass that down to `EnsureDownloaded` via lambda.

Reply:
It just looks like a code smell, it feels wrong. There is this implicit assumption that we're only setting this field once, that we're never accessing the field before we set it, almost like a constructor. T